### PR TITLE
add remote tso support

### PR DESCRIPTION
**Pull request by disksing**

### CODE REVIEW

**Comment by siddontang** [comment]
The package name is oracle, so using LocalOracle or TsoOracle as the struct name seems redundant.
see https://github.com/golang/go/wiki/CodeReviewComments#package-names

**Comment by disksing** [comment]
Good point. How about using `Local` and `Tso`? @siddontang

**Comment by siddontang** [comment]
Btw, I am confused why using oracle as the package name too. In themis, it uses timestamp directly.  
I don't think Tso is good too, Ts should be TS, but I haven't found another better name.

**Comment by c4pt0r** [comment]
oracle is the original name in paper (percolator) and themis.

**Comment by c4pt0r** [comment]
In themis, the class name is:
public abstract class BaseTimestampOracle {
    ...
}

**Comment by siddontang** [comment]
Tso is short for Timestamp oracle, I prefer RemoteOracle or ProxyOracle.

**Comment by disksing** [comment]
OK. I'll keep `LocalOracle` and rename `TsoOracle` to `RemoteOracle`. @siddontang @c4pt0r

File: oracle/oracles/remote.go [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+func NewRemoteOracle(addr string) *RemoteOracle {
+	return &RemoteOracle{
+		c: client.NewClient(&client.Conf{ServerAddr: addr}),
+	}
+}
+
+func (t *RemoteOracle) GetTimestamp() (uint64, error) {
+	ts, err := t.c.GoGetTimestamp().GetTS()
+	if err != nil {
+		return 0, err
```
Comment (qiuyesuifeng) [inline oracle/oracles/remote.go]: errors.Trace

File: oracle/oracle.go [hunk 2]
```diff
@@ -1,5 +1,5 @@
 package oracle
 
 type Oracle interface {
-	GetTimestamp() uint64
+	GetTimestamp() (uint64, error)
```
Comment (qiuyesuifeng) [inline oracle/oracle.go]: I think it is better to add a retry mechanism.

File: oracle/oracles/remote.go [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+	c *client.Client
+}
+
+func NewRemoteOracle(addr string) *RemoteOracle {
+	return &RemoteOracle{
+		c: client.NewClient(&client.Conf{ServerAddr: addr}),
+	}
+}
+
+func (t *RemoteOracle) GetTimestamp() (uint64, error) {
```
Comment (qiuyesuifeng) [inline oracle/oracles/remote.go]: I think it is better to add a retry mechanism.

**Comment by siddontang** [comment]
You can supply NewLocalOracle for outer use like NewRemoteOracle too.

**Comment by siddontang** [comment]
LGTM

**Pull request merged**